sgt-back-end/index.js

```diff
@@ -0,0 +1,140 @@
+const express = require('express');
+const app = express();
+app.use(express.json());
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
+      res.status(200).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'Query has failed.'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const data = req.body;
+  const sql = {
+    text: 'INSERT INTO grades(name, course, score) VALUES($1, $2, $3) RETURNING *',
+    values: [data.name, data.course, data.score]
+  };
```

Comment:
We typically validate the data, then build the SQL query. This avoids the server doing unnecessary work with any of the data is missing or bad.